aptos-move/framework/aptos-framework/sources/object.spec.move

```diff
@@ -115,48 +213,111 @@ spec aptos_framework::object {
-        let bytes = concat(bytes_spec,vec<u8>(OBJECT_FROM_GUID_ADDRESS_SCHEME));
+        let bytes = concat(bytes_spec, vec<u8>(OBJECT_FROM_GUID_ADDRESS_SCHEME));
         let hash_bytes = hash::sha3_256(bytes);
         let obj_addr = from_bcs::deserialize<address>(hash_bytes);
         aborts_if exists<ObjectCore>(obj_addr);
         aborts_if !from_bcs::deserializable<address>(hash_bytes);
+
+        ensures global<ObjectCore>(addr).guid_creation_num == old(global<ObjectCore>(addr)).guid_creation_num + 1;
+        ensures exists<ObjectCore>(obj_addr);
+        ensures global<ObjectCore>(obj_addr) == ObjectCore {
+                guid_creation_num: INIT_GUID_CREATION_NUM + 1,
+                owner: addr,
+                allow_ungated_transfer: true,
+                transfer_events: event::EventHandle {
+                    counter: 0,
+                    guid: guid::GUID {
+                        id: guid::ID {
+                            creation_num: INIT_GUID_CREATION_NUM,
+                            addr: obj_addr,
+                        }
+                    }
+                }
+        };
+        ensures result == ConstructorRef { self: obj_addr, can_delete: true };
     }
 
     spec create_object_from_guid(creator_address: address, guid: guid::GUID): ConstructorRef {
         let bytes_spec = bcs::to_bytes(guid);
-        let bytes = concat(bytes_spec,vec<u8>(OBJECT_FROM_GUID_ADDRESS_SCHEME));
+        let bytes = concat(bytes_spec, vec<u8>(OBJECT_FROM_GUID_ADDRESS_SCHEME));
         let hash_bytes = hash::sha3_256(bytes);
         let obj_addr = from_bcs::deserialize<address>(hash_bytes);
         aborts_if exists<ObjectCore>(obj_addr);
         aborts_if !from_bcs::deserializable<address>(hash_bytes);
+
+        ensures exists<ObjectCore>(obj_addr);
+        ensures global<ObjectCore>(obj_addr) == ObjectCore {
+                guid_creation_num: INIT_GUID_CREATION_NUM + 1,
+                owner: creator_address,
+                allow_ungated_transfer: true,
+                transfer_events: event::EventHandle {
+                    counter: 0,
+                    guid: guid::GUID {
+                        id: guid::ID {
+                            creation_num: INIT_GUID_CREATION_NUM,
+                            addr: obj_addr,
+                        }
+                    }
+                }
+        };
+        ensures result == ConstructorRef { self: obj_addr, can_delete: true };
     }
 
     spec create_object_internal(
         creator_address: address,
         object: address,
         can_delete: bool,
     ): ConstructorRef {
+        // property 1: Creating an object twice on the same address must never occur.
         aborts_if exists<ObjectCore>(object);
+        ensures exists<ObjectCore>(object);
+        // property 6: Object addresses must not overlap with other addresses in different domains.
+        ensures global<ObjectCore>(object) == ObjectCore {
```

Comment:
For property 6, I think you can just specify something like global<ObjectCore>(object).guid_creation_num == INIT_GUID_CREATION_NUM + 1;

Reply:
yes, it's better this way, fixed